### PR TITLE
chore(deps): Downgrade typescript & jest to versions used in sentry-javascript

### PR DESCRIPTION
**Pull request by mydea**
For reasons™️ we are stuck with an old version of typescript (3.8) in sentry-javascript.
In order to prepare the migration there, this PR downgrades typescript to that version as well.
With it, that required some other changes, mainly also downgrading jest, which required adapting some tests... 😢 not ideal, but I tried to make it work with as little changes as possible.

The two biggest issues were:

* jest 27 does not support `jest.useFakeTimers({ autoAdvance: true })`, so needed to work around that somehow. I tried to replicate this manually
* jest 27 does not support `expect.closeTo()`. This was used once, I commented this out for now.

Other things:

* `@ts-expect-error` is not supported, so instead we use `@ts-ignore`
* eslint also needed downgrading
* I had issues with `@jest/globals` (which we also don't use in sentry-javascript), so I changed this to use the globals everywhere

### CODE REVIEW

**Comment by mydea** [comment]
@billyvg maybe you could take a look at the failing test? I am lacking a bit of context on that test (e.g. what exactly the timing of stuff _should_ be etc), so having a hard time debugging this 😅 also the "replacement" for the `autoAdvance` stuff is def. a bit "hacky", so maybe needs adjustment for specific tests I guess...

**Comment by mydea** [comment]
I think the `size` failed CI job can be ignored, that's because master has a different ts version leading to issues.

**Comment by billyvg** [comment]
Weird, tests are passing locally for me

File: src/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare global {
+  interface PerformancePaintTiming {
+    a: number;
+  }
+}
```
[the remark this answers is not in the document]
Comment (mydea) [reply src/global.d.ts:5]: leftover from debugging xD oops

File: src/index-sampling.test.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+jest.useFakeTimers();
+
+let interval: any;
+beforeAll(function () {
+  interval = _setInterval(() => jest.advanceTimersByTime(20), 20);
+});
+
+afterAll(function () {
+  _clearInterval(interval);
+});
```
[the remark this answers is not in the document]
Comment (mydea) [reply src/index-sampling.test.ts:17]: yeah, I also thought so. When everything works, I'll extract this into a method!

**Comment by mydea** [comment]
Tests seem to be passing now! Had to do some further adjustments for the build as well, but _seems_ all good now.

File: src/index-sampling.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mydea) [reply src/index-sampling.test.ts]: done! ✅

**Comment by mydea** [comment]
> This looks good to me -- can we separate out removing `@jest/globals` into its own PR? I would like to have it in its own commit so it's easy to reference/revert down the road.

👍 will do! see: https://github.com/getsentry/sentry-replay/pull/319

**Review by billyvg — APPROVED** [review]

**Pull request merged**